specification/dartLangSpec.tex

```diff
@@ -3154,11 +3158,44 @@ \subsubsection{Generative Constructors}
 <initializerExpression> ::= <conditionalExpression> | <cascade>
 \end{grammar}
 
+\LMHash{}%
+If an \synt{initializerExpression} is a \synt{functionExpression},
+a compile-time error occurs.
```

Comment:
That's sadly not enough.

We don't allow function expressions at "top level", meaning they also cannot occur as operator operands. They have to be inside parentheses to be allowed.
Examples that are not allowed:
```dart
class C {
  var x, y;
  C() : x = a + () {}, y = () {} + a;
}
```
which could otherwise be made valid by a suitable definition of `a` or an extension method, like:
```dart
extension on void Function() { 
  operator +(void Function() f) => () { this(); f(); }
}
void a() { print("a"); }
``` 
So, the description needs to be more elaborate.

Maybe: 
> It is a compile-time error if an initializerExpression contains a functionExpression which is not inside a parethesizedExpression.

Reply:
Right, done.